app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,125 @@
+import { ObservableStore } from '@metamask/obs-store';
+import MetaMetricsController from '../metametrics';
+import {
+  createDataDeletionRegulationTask,
+  fetchDeletionRegulationStatus,
+} from './services/services';
```

Comment:
Nit: The way we have been building external API service modules recently is to construct them as classes, then pass them into the controller as constructor parameters (either that or use the messenger to communicate with them, which is planned but hasn't been done yet, see https://github.com/MetaMask/decisions/blob/main/decisions/core/0001-messaging-non-controllers.md)

Injecting the services rather than importing them would make it easier to mock out the service in unit tests